### ISSUE TITLE
Change some global colors or extend the FlatLaf / Change button's background

### ISSUE THREAD

**Issue by tsjung75** [JFormDesigner/FlatLaf#116]
Hello,

I've found the FlatLaf and it seems to be great look and feel.

BTW, I want to change some colors such as selectionBackground in the FlatLaf.
I tried with UIManager.put("selectionBackground", color) or similar things, but it didn't work.
Of course, individual component color such as UIManager.put("Table.selectedBackground", color) worked.

I have two questions (Goals are almost same)

1. How to change the global selectionBackground (or selectedBackground)?
2. How to extend FlatLaf and add a custom .properties file and use it?

One more question.

When setting the button's colors (only a few for special purpose. not global)
- setForeground(Color.WHITE), setBackground(Color.BLUE),
If the button lose the focus, the foreground color is changed to BLACK.
This effect is not what I intend. (I expect the foreground must be changed to grayed WHITE)
How to accomplish my intention?
Do I need to set the client property for non-focused foreground in the button?

Thanks in advance.

**Comment by DevCharly** [comment]
There is no global `selectionBackground` value in the UI defaults. You have to set them for individual components. 

Here is a list of UI default values:
https://github.com/JFormDesigner/FlatLaf/blob/master/flatlaf-testing/src/main/resources/com/formdev/flatlaf/testing/uidefaults/FlatLightLaf_1.8.0_202.txt

In a own FlatLaf properties file you can change the global selection color with:
~~~
@selectionBackground=#f00
~~~

There are three ways to customize FlatLaf with own .properties file.

## A) use class [FlatPropertiesLaf](https://javadoc.io/doc/com.formdev/flatlaf/latest/com/formdev/flatlaf/FlatPropertiesLaf.html)

(see linked javadoc for details)

## B) create own LaF by extending FlatLaf class

You need a Java class, which you later use in `UIManager.setLookAndFeel()`:
~~~java
package my.pkg;

public class MyDarkLaf
    extends FlatDarkLaf
{
    @Override
    public String getName() {
        return "My Dark Laf";
    }
}
~~~

In the same package as the class create a properties files with the same name as the class.
E.g. `MyDarkLaf.properties`

This is similar to A), but in B) you have a LaF class with a no-param constructor, which can be useful when supporting multiple LaFs in the app.

## C) implement a service

You need a service class that extends class [FlatDefaultsAddon](https://javadoc.io/doc/com.formdev/flatlaf/latest/com/formdev/flatlaf/FlatDefaultsAddon.html):
~~~java
package my.pkg;

public class MyFlatLafAddon extends com.formdev.flatlaf.FlatDefaultsAddon {}
~~~

And you need a file named `META-INF/services/com.formdev.flatlaf.FlatDefaultsAddon` in your JAR/classpath. Content is a single line with the classname of your service class:
~~~
my.pkg.MyFlatLafAddon
~~~

Then you can create several .properties files in the same package as the service class. The names must be `FlatLaf.properties`, `FlatLightLaf.properties`, `FlatDarkLaf.properties`, `FlatIntelliJLaf.properties` or `FlatDarculaLaf.properties`. All files are optional.

This allows modifying multiple FlatLaf themes.

If it seems not to work, override the method `getDefaults()` of the service class and check with a breakpoint whether it is invoked.

**Comment by DevCharly** [comment]
Regarding button color:
If a button is **unfocused**, `JButton.getForeground()` and `getBackground()` are used.
But if it is **focused**, different colors (specified by the Laf) are used.

Unfocused (with explicit colors):
![image](https://user-images.githubusercontent.com/5604048/85435474-cba4a980-b587-11ea-9541-debe7366428b.png)

Focused (with same explicit colors as above):
![image](https://user-images.githubusercontent.com/5604048/85435515-d8290200-b587-11ea-95fd-e0423f5070a1.png)

I agree that this is far from optimal and consider changing this for the next release so that explicit colors are also used for focused buttons. Thanks for pointing out this issue.

**Comment by tsjung75** [comment]
> There is no global `selectionBackground` value in the UI defaults. You have to set them for individual components.
> 
> Here is a list of UI default values:
> https://github.com/JFormDesigner/FlatLaf/blob/master/flatlaf-testing/src/main/resources/com/formdev/flatlaf/testing/uidefaults/FlatLightLaf_1.8.0_202.txt
> 
> In a own FlatLaf properties file you can change the global selection color with:
> 
> ```
> @selectionBackground=#f00
> ```
> 
> There are three ways to customize FlatLaf with own .properties file.
> 
> ## A) use class [FlatPropertiesLaf](https://javadoc.io/doc/com.formdev/flatlaf/latest/com/formdev/flatlaf/FlatPropertiesLaf.html)
> (see linked javadoc for details)
> 
> ## B) create own LaF by extending FlatLaf class
> You need a Java class, which you later use in `UIManager.setLookAndFeel()`:
> 
> ```java
> package my.pkg;
> 
> public class MyDarkLaf
>     extends FlatDarkLaf
> {
>     @Override
>     public String getName() {
>         return "My Dark Laf";
>     }
> }
> ```
> 
> In the same package as the class create a properties files with the same name as the class.
> E.g. `MyDarkLaf.properties`
> 
> This is similar to A), but in B) you have a LaF class with a no-param constructor, which can be useful when supporting multiple LaFs in the app.
> 
> ## C) implement a service
> You need a service class that extends class [FlatDefaultsAddon](https://javadoc.io/doc/com.formdev/flatlaf/latest/com/formdev/flatlaf/FlatDefaultsAddon.html):
> 
> ```java
> package my.pkg;
> 
> public class MyFlatLafAddon extends com.formdev.flatlaf.FlatDefaultsAddon {}
> ```
> 
> And you need a file named `META-INF/services/com.formdev.flatlaf.FlatDefaultsAddon` in your JAR/classpath. Content is a single line with the classname of your service class:
> 
> ```
> my.pkg.MyFlatLafAddon
> ```
> 
> Then you can create several .properties files in the same package as the service class. The names must be `FlatLaf.properties`, `FlatLightLaf.properties`, `FlatDarkLaf.properties`, `FlatIntelliJLaf.properties` or `FlatDarculaLaf.properties`. All files are optional.
> 
> This allows modifying multiple FlatLaf themes.
> 
> If it seems not to work, override the method `getDefaults()` of the service class and check with a breakpoint whether it is invoked.

Thanks for the detailed information.
I will do as you guided

**Comment by DevCharly** [comment]
Regarding button color:
I've changed behavior in commit 6914a6132ceb161aa62f860e69c5a0ca3b46cdde slightly: explicitly set background or foreground colors are now used even if button is focused

**Comment by tsjung75** [comment]
I tried what you committed. It works well :) Thank you.

**Comment by DevCharly** [comment]
this is now released in [FlatLaf 0.37](https://github.com/JFormDesigner/FlatLaf/releases/tag/0.37)

**Comment by tsjung75** [comment]
Sorry. I meant another thing. 
JToggleButton also works like JButton.

BTW, for disabled status is this same? 
When I using white foreground color for button in disabled status, I expect the foreground color close to the white.

Actually, for all components (such as JList, JTable, etc), is this same?
